File: backend/src/main/java/com/nutriai/api/model/Scan.java
package com.nutriai.api.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "scans")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Scan {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(name = "image_url", length = 1000)
    private String imageUrl;

    @Column(name = "food_name", nullable = false)
    private String foodName;

    private Double calories = 0.0;
    private Double carbs = 0.0;
    private Double protein = 0.0;
    private Double fat = 0.0;

    @Column(name = "weight_grams")
    private Double weightGrams = 100.0;

    @Column(name = "health_score")
    private Integer healthScore = 5;

    @Column(name = "health_advice", columnDefinition = "TEXT")
    private String healthAdvice;

    @Column(name = "created_at")
    private LocalDateTime createdAt = LocalDateTime.now();

    @OneToMany(mappedBy = "scan", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<FoodDetail> ingredients = new ArrayList<>();
}

File: backend/app/models/scan.py
from datetime import datetime, timezone
from sqlalchemy import Column, Integer, String, Float, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from app.db.session import Base

class Scan(Base):
    __tablename__ = "scans"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id", ondelete="CASCADE"), nullable=False)
    image_url = Column(String, nullable=True)
    food_name = Column(String, nullable=False, index=True)
    
    # Dinh dưỡng đa lượng
    calories = Column(Float, default=0.0)
    carbs = Column(Float, default=0.0)
    protein = Column(Float, default=0.0)
    fat = Column(Float, default=0.0)
    weight_grams = Column(Float, default=100.0)
    
    # Điểm đánh giá và lời khuyên sức khỏe
    health_score = Column(Integer, default=5)
    health_advice = Column(String, nullable=True)
    
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc))

    # Mối quan hệ
    user = relationship("User", back_populates="scans")
    ingredients = relationship("FoodDetail", back_populates="scan", cascade="all, delete-orphan")
